vcd/resource_vcd_vapp_vm.go

```diff
@@ -616,9 +778,33 @@ func resourceVcdVAppVmRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.Set("name", vm.VM.Name)
-	if len(vm.VM.NetworkConnectionSection.NetworkConnection) > 0 {
+	networks := d.Get("networks").([]interface{})
+	network := d.Get("network_name").(string)
+	switch {
+	// network_name is not set. networks is set in config
+	case network != "":
 		d.Set("ip", vm.VM.NetworkConnectionSection.NetworkConnection[0].IPAddress)
+		d.Set("mac", vm.VM.NetworkConnectionSection.NetworkConnection[0].MACAddress)
+	// We are using networks block and rebuilding statefile
+	case len(networks) > 0:
+		var nets []map[string]interface{}
+		// Loop over existing NICs in VM
+		for i, vmNet := range vm.VM.NetworkConnectionSection.NetworkConnection {
+			singleNIC := make(map[string]interface{})
+			singleNIC["ip"] = vmNet.IPAddress
+			singleNIC["mac"] = vmNet.MACAddress
+			singleNIC["orgnetwork"] = vmNet.Network
+			singleNIC["ip_allocation_mode"] = vmNet.IPAddressAllocationMode
+
+			singleNIC["is_primary"] = false
+			if i == vm.VM.NetworkConnectionSection.PrimaryNetworkConnectionIndex {
+				singleNIC["is_primary"] = true
+			}
+			nets = append(nets, singleNIC)
+		}
+		d.Set("networks", nets)
```

Comment:
I know I proposed this, but I spotted it is missing one more feature.
Could you sort the `nets` by NetworkConnectionIndex. What happens now is that the statefile does not necessarily gets update with the right order of network interfaces.

<img width="309" alt="image" src="https://user-images.githubusercontent.com/15804230/56035332-8d301f80-5d32-11e9-87ff-34edffa98a24.png">
 
```
"networks.#": "4",
                            "networks.0.ip": "11.10.0.153",
                            "networks.0.ip_allocation_mode": "POOL",
                            "networks.0.is_primary": "false",
                            "networks.0.mac": "00:50:56:01:03:d2",
                            "networks.0.orgnetwork": "multinic-net",
                            "networks.1.ip": "",
                            "networks.1.ip_allocation_mode": "DHCP",
                            "networks.1.is_primary": "true",
                            "networks.1.mac": "00:50:56:01:03:d3",
                            "networks.1.orgnetwork": "multinic-net",
                            "networks.2.ip": "12.10.0.152",
                            "networks.2.ip_allocation_mode": "POOL",
                            "networks.2.is_primary": "false",
                            "networks.2.mac": "00:50:56:01:03:d4",
                            "networks.2.orgnetwork": "multinic-net2",
                            "networks.3.ip": "11.10.0.152",
                            "networks.3.ip_allocation_mode": "POOL",
                            "networks.3.is_primary": "false",
                            "networks.3.mac": "00:50:56:01:03:d1",
                            "networks.3.orgnetwork": "multinic-net",
```

If I do a `terraform apply` with zero it then suggests to recreate NICs because of incorrect order.